rest/blip_sync.go

```diff
@@ -903,18 +885,19 @@ func (bh *blipHandler) handleRev(rq *blip.Message) error {
 		//       while retrieving deltaSrcRevID.  Couchbase Lite replication guarantees client has access to deltaSrcRevID,
 		//       due to no-conflict write restriction, but we still need to enforce security here to prevent leaking data about previous
 		//       revisions to malicious actors (in the scenario where that user has write but not read access).
-		deltaSrcBody, err := bh.db.GetRevCopy(docID, deltaSrcRevID, false, nil, db.BodyDeepCopy)
+		deltaSrcRev, err := bh.db.GetRev(docID, deltaSrcRevID, false, nil)
```

Comment:
Verify handling for the case where deltaSrcRev.deleted = true.  CBL shouldn't be sending revisions where the source revision is a tombstone, but should have defensive handling in SG (and need to make sure that the switch to use MutableBody below isn't going to break in a new way).